drivers: stm32: always call runtime_get when registering targets

fabiobaltieri opened the pull request:
Current code only calls pm_device_runtime_get when registering target devices if the device is marked as wakeup capable by the application.

This does quite work for a setup with PM=n and PM_DEVICE=y and PM_DEVICE_RUNTIME=y, where the CPU does not really go in stop mode ever, and results in the i2c target device not working at all.

Fix this by always claiming the device when PM_DEVICE_RUNTIME is enabled.

fabiobaltieri committed:
i2c: stm32: always call runtime_get when registering targets

Current code only calls pm_device_runtime_get when registering target
devices if the device is marked as wakeup capable by the application.

This does quite work for a setup with PM=n and PM_DEVICE=y and
PM_DEVICE_RUNTIME=y, where the CPU does not really go in stop mode ever,
and results in the i2c target device not working at all.

Fix this by always claiming the device when PM_DEVICE_RUNTIME is
enabled.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri commented:
cc @wouterh

bjarki-andreasen reviewed (dismissed):
Looks correct :) If you want, you can remove the ifdefs
```
#if defined(CONFIG_PM_DEVICE_RUNTIME)
```
since the functions automatically are optimized out in these cases (they have empty mocks)

fabiobaltieri committed:
i2c: stm32: drop few redundant guards

Drop few CONFIG_PM_DEVICE_RUNTIME guards, the pm_device_runtime
functions they are masking are no-op automatically when the
corresponding config option is not selected.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri commented:
Okay thanks for explaining, need to wrap my head about the wakup source but whatever. Added a new commit to remove the guards, I think they are all unnecessary, as pointed out both the pm runtime and the wakeup functions have noops.

bjarki-andreasen reviewed (approved):
Definite improvement I believe :)
